packages/core/src/components/fields/ObjectField.tsx

```diff
@@ -80,7 +80,10 @@ class ObjectField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends Fo
       }
       // Copy the current path and insert in the property name into the first location
       const changePath = Array.isArray(path) ? path.slice() : [];
-      changePath.unshift(name);
+      // Ensure we are not duplicating the path
+      if (!changePath.length || changePath[0] !== name) {
```

Comment:
Since there is similar code in `ArrayField` might I consider making a similar change there?